tests/test_deployments/test_app.py

```diff
@@ -271,3 +279,78 @@ def get_missing(messages: List[str]) -> List[str]:
         assert not get_missing(before_stopping), get_logs()
         self.tendermint_node.stop()
         assert not get_missing(after_stopping), get_logs()
+
+
+def mock_get_node_command_kwargs() -> Dict:
+    """Get the node command kwargs"""
+    kwargs = {
+        "bufsize": 1,
+        "universal_newlines": True,
+    }
+
+    # Pipe stdout and stderr even if we're not going to read to provoke the buffer fill
+    kwargs["stdout"] = subprocess.PIPE
+    kwargs["stderr"] = subprocess.STDOUT
+
+    if platform.system() == "Windows":  # pragma: nocover
+        kwargs["creationflags"] = subprocess.CREATE_NEW_PROCESS_GROUP  # type: ignore
+    else:
+        kwargs["preexec_fn"] = os.setsid  # type: ignore
+
+    return kwargs
+
+
+class TestTendermintBufferFailing(BaseTendermintServerTest):
+    """Test Tendermint buffer"""
+
+    perform_monitoring = False  # Setting this flag to False and not monitoring makes the buffer to fill and freeze Tendermint
+    debug_tendermint = True  # This will cause more logging -> faster failure
+
+    @wait_for_node_to_run
+    @mock.patch(
+        "autonomy.data.Dockerfiles.tendermint.tendermint.TendermintParams.get_node_command_kwargs",
+        mock_get_node_command_kwargs,
+    )
```

Comment:
This patch is not working. I've tried several approaches, but it seems that the node does not exists at this point so I can't use mock.patch.object. Any idea @Karrenbelt ?

Reply:
can try something like 
```
    @classmethod
    def setup_class(cls) -> None:
        with mock.patch.object(TendermintParams, "get_node_command_kwargs", return_value=mock_get_node_command_kwargs()):
            super().setup_class()
```